framework/db/src/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -1927,6 +1935,54 @@ public Integer getDistinctCount(SearchCriteria<T> sc) {
         }
     }
 
+    public Integer getDistinctCount(SearchCriteria<T> sc, String[] distinctColumns) {
+        String clause = sc != null ? sc.getWhereClause() : null;
+        if (clause != null && clause.length() == 0) {
```

Comment:
This can be replaced with `Strings.isNullOrEmpty()`

Reply:
Replaced as suggested.